1 to 1 reqs

KyraWayne opened the pull request:
Styling for 1 to 1 reqs

Commit:
Merge branch 'master' into 1-to-1-reqs

jkingsman committed:
Add classname to satisfier dropdowns with less than two entries

Commit:
replace class dropdown for checkbox for 1:1 classes

organizes req checkbox, req dropdown and class dropdown in new div
reqEntryDiv. only displays checkbox if list is length of json lists is
over 3

Commit:
Merge branch 'master' into 1-to-1-reqs

Commit:
Merge branch 'master' into 1-to-1-reqs

KyraWayne commented:
@jkingsman

jkingsman commented on js/classloader.js:
```diff
@@ -71,21 +71,32 @@ function fillColumn(colNumber) {
             requirementCheckbox.id = requirementName;
             requirementCheckbox.className = classes + ' pull-right requirementMarker';
-            requirementCheckbox.style.display = 'none';
+            requirementCheckbox.style.marginRight = '5px';
```
Can we add a class and put it in the stylesheet? In-JS styles are really hard to manage/track down where styling is coming from.

jkingsman commented on js/classloader.js:
```diff
@@ -71,21 +71,32 @@ function fillColumn(colNumber) {
+
+            if(classSatisfactionDropdown.length > 2){
+                reqEntryDiv.appendChild(classSatisfactionDropdown);
+                requirementCheckbox.style.display = 'none';
```
Same as above - can we put this into a class?

jkingsman commented on js/helpers.js:
```diff
@@ -222,7 +222,7 @@ function restoreStatus() {
 
         requirementEntry.appendChild(requirementLabel);
-        requirementEntry.appendChild(document.createElement('br'));
+       // requirementEntry.appendChild(document.createElement('br'));
```
If we don't need this, just go ahead and delete it -- no need to comment code unless it's going to be used later.

jkingsman reviewed (changes requested):
Looks great conceptually - just a couple places to move styling out of the JS and into CSS and a comment deletion.

Commit:
uses CSS styling rather than JS

deletes unnecessary commented out items and adds classes to move
styling to style.css instead

Commit:
Merge branch 'master' into 1-to-1-reqs

KyraWayne commented:
@jkingsman see new commits

jkingsman commented on js/classloader.js:
```diff
@@ -71,21 +71,30 @@ function fillColumn(colNumber) {
+
+            if(classSatisfactionDropdown.length > 2){
+                reqEntryDiv.classList.add('hasClassDropdown');
+                reqEntryDiv.appendChild(classSatisfactionDropdown);
```
Can we append no matter what and just have it set to `display: none`? It breaks some other stuff that has to do with knowing when to and when not to display the satisfying class by the requirement in the calendar (note all the undefineds now).

Commit:
adds new class hasCheckbox

complements hasClassDropdown

jkingsman commented:
Looks great; awesome.
